src/NuGet.Core/NuGet.Protocol/Providers/PluginResourceProvider.cs

```diff
@@ -21,57 +21,20 @@ namespace NuGet.Protocol.Core.Types
     /// A plugin resource provider.
     /// </summary>
     /// <remarks>This is unsealed only to facilitate testing.</remarks>
-    public class PluginResourceProvider : ResourceProvider, IDisposable
+    public class PluginResourceProvider : ResourceProvider
     {
-        private const string _idleTimeoutEnvironmentVariable = "NUGET_PLUGIN_IDLE_TIMEOUT_IN_SECONDS";
-        private const string _pluginPathsEnvironmentVariable = "NUGET_PLUGIN_PATHS";
 
-        private ConnectionOptions _connectionOptions;
-        private Lazy<IPluginDiscoverer> _discoverer;
-        private bool _isDisposed;
-        private IPluginFactory _pluginFactory;
-        private ConcurrentDictionary<PluginPackageSourceKey, Lazy<Task<IReadOnlyList<OperationClaim>>>> _pluginOperationClaims;
-        private ConcurrentDictionary<string, Lazy<IPluginMulticlientUtilities>> _pluginUtilities;
-        private string _rawPluginPaths;
+        private readonly PluginManager _pluginManager;
```

Comment:
You removed `IDisposable` from `PluginResourceProvider`'s base list, but the class owns `_pluginManager`, which implements `IDisposable`.

Reply:
The PluginManager is now singleton. 
IDisposable doesn't quite work here. 
I'll get back to you on this.

Reply:
I've made the changes as per our offline discussion. 
IPluginManager is now referenced everywhere in product code to hide that PluginManager is disposable.
Cleaning up some extra logic before deeming this as ready for review again.